src/main/drivers/flash_m25p16.c

```diff
@@ -179,52 +174,41 @@ static bool m25p16_waitForReady(flashDevice_t *fdevice)
 
 bool m25p16_detect(flashDevice_t *fdevice, uint32_t chipID)
 {
-    switch (chipID) {
-    case JEDEC_ID_WINBOND_W25Q16:
-    case JEDEC_ID_MICRON_M25P16:
-        fdevice->geometry.sectors = 32;
-        fdevice->geometry.pagesPerSector = 256;
-        break;
-    case JEDEC_ID_BERGMICRO_W25Q32:
-        fdevice->geometry.sectors = 1024;
-        fdevice->geometry.pagesPerSector = 16;
-        break;
-    case JEDEC_ID_WINBOND_W25Q32:
-    case JEDEC_ID_MACRONIX_MX25L3206E:
-        fdevice->geometry.sectors = 64;
-        fdevice->geometry.pagesPerSector = 256;
-        break;
-    case JEDEC_ID_MICRON_N25Q064:
-    case JEDEC_ID_WINBOND_W25Q64:
-    case JEDEC_ID_MACRONIX_MX25L6406E:
-        fdevice->geometry.sectors = 128;
-        fdevice->geometry.pagesPerSector = 256;
-        break;
-    case JEDEC_ID_MICRON_N25Q128:
-    case JEDEC_ID_WINBOND_W25Q128:
-    case JEDEC_ID_WINBOND_W25Q128_DTR:
-    case JEDEC_ID_CYPRESS_S25FL128L:
-        fdevice->geometry.sectors = 256;
-        fdevice->geometry.pagesPerSector = 256;
-        break;
-    case JEDEC_ID_WINBOND_W25Q256:
-    case JEDEC_ID_MACRONIX_MX25L25635E:
-        fdevice->geometry.sectors = 512;
-        fdevice->geometry.pagesPerSector = 256;
-        break;
+    flashGeometry_t *geometry = &fdevice->geometry;
+
+    switch(chipID) {
+        JEDEC_CLKS(MACRONIX_MX25L3206E);
```

Comment:
This will make all syntax checkers / highlighters go haywire - can you please change this so the syntax of the `switch` statement is preserved?

Reply:
Switched to using a table which is much easier to expand should we wish to add new devices.